### ISSUE TITLE
Hash should also depend on os version

### ISSUE THREAD

**Issue by sagudev** [baptiste0928/cargo-install#20]
At servo we are using same workflow on ubuntu 22.04 for PR builds and 20.04 on nightly builds. Because the cache has version from 22.04 cached version does not work on 20.04 (due to different version of GLIBC). Hash must not only depend on runner os, but also on version of OS (at least on linux).

See https://github.com/servo/servo/issues/30835

**Comment by baptiste0928** [comment]
Hi, thanks for reporting this. I'll take a look at including the OS version in the cache key when I get a moment.

In the meantime, you can provide a custom `cache-key` that will be added to the strings used to compute the final cache key:

```yaml
 - name: Install taplo
   uses: baptiste0928/cargo-install@v2
     with:
       crate: taplo-cli
       cache-key: 'custom-key'
```

**Comment by baptiste0928** [comment]
I just released [v3.0.0](https://github.com/baptiste0928/cargo-install/releases/tag/v3.0.0), which now add the current os version to the cache key (see https://github.com/baptiste0928/cargo-install/commit/313e40291ae18253c74cab6ad775cd8471b6d0eb for the implementation).

You will need to update your workflows because of the major version (I updated the action to use Node 20 instead of Node 16 at the same time):

```diff
-uses: baptiste0928/cargo-install@v2
+uses: baptiste0928/cargo-install@v3
```
